compiler/evaltempl.nim

```diff
@@ -80,9 +80,14 @@ proc evalTemplateArgs(n: PNode, s: PSym; fromHlo: bool): PNode =
     expectedRegularParams = <s.typ.len
     givenRegularParams = totalParams - genericParams
   if givenRegularParams < 0: givenRegularParams = 0
+
   if totalParams > expectedRegularParams + genericParams:
     globalError(n.info, errWrongNumberOfArguments)
 
+  if totalParams < genericParams:
+    globalError(n.info, errMissingGenericParamsForTemplate,
```

Comment:
in the future, consider that ``globalError`` is really bad for ``nimsuggest``, it's better to use ``localError`` and emit an error node instead. (I know the existing code above also uses globalError.)